src/core/exec-invoke.c

```diff
@@ -2280,6 +2280,56 @@ static int create_many_symlinks(const char *root, const char *source, char **sym
         return 0;
 }
 
+static int is_idmapping_supported(const char *path) {
```

Comment:
i would prefer if this addition was split into a separate, preparatory commit.

and if it would be placed in namespace-util.[ch] as it is reasonably generic.

and if it would come with a superficial test case that just runs this on a few well-known paths and shows the results (without encoding expecations, after all our tests need to run on super old kernels)

Reply:
made this check and added the superficial tests to test-namespace.c, let me know if this is the right place and if the sanity checks there are ok. Were you thinking of any other paths?